src/panels/config/devices/ha-config-devices-dashboard.ts

```diff
@@ -338,11 +338,24 @@ export class HaConfigDeviceDashboard extends LitElement {
               : undefined;
           const batteryIsBinary =
             battery && computeStateDomain(battery) === "binary_sensor";
+
+          let battery_state_display = battery?.state;
+          if (
+            typeof battery?.state === "string" &&
+            battery.state.trim() &&
+            !isNaN(Number(battery.state))
+          ) {
+            // Remove trailing 0s from battery decimal
+            battery_state_display = Number(battery.state).toString();
+          }
+
           return battery && (batteryIsBinary || !isNaN(battery.state as any))
             ? html`
                 ${batteryIsBinary
                   ? ""
-                  : battery.state + blankBeforePercent(this.hass.locale) + "%"}
```

Comment:
I think all you need to do is replace `battery.state` here with the rounding.  Your `if` block is just repeating the conditionals.

I would also use `Number(battery.state).toFixed()` instead.  I think any decimal are unnecessary in the table view.  The more-info dialog can show higher precision.

Reply:
Updated.